### ISSUE TITLE
Series Sorting naming issue on Linux

### ISSUE THREAD

**Issue by Annihilator6000** [sabnzbd/sabnzbd#2068]
OS: Ubuntu Server 18.04.6 LTS (GNU/Linux 4.15.0-167-generic x86_64)
SABnzbd version: 3.5.0 (via jcfp/nobetas repo) - also affects v3.4.2 and possibly v3.4.1/v3.4.0.

#### Issue:
  SABnzbd improperly renames TV show files during series sorting causing short file names to be shown in other operating systems that access the server via a Samba network share.

#### Details:
  Something changed with Series Sorting after v3.4.X was released that causes TV show files to be named incorrectly on Ubuntu Server and possibly other Linux distros. A backslash is joined in between the show name and the file name. 
  
  I've observed this issue in versions 3.5.0 and 3.4.2. I think it was happening in earlier 3.4.X versions as well, but I can't be completely sure without some testing. I believe that this issue might coincide with the implementation of GuessIt. Movie Sorting still works properly though. Because of this issue I have been reverting back to using v3.3.1 where Series Sorting still works properly.

I don't know if it makes any difference with the newer versions that use GuessIt or not, but the sorting string I use is: %sn\%dn.%ext (Show Name\Original File Name S01E01.ext).

I uploaded a section of the log for a test download that I did that has this issue: https://zerobin.net/?c25eacead648e650#dG7PZp6siKrZTdRREWwz6N3d59VLRshRWNDo1VoGUNY=
pw: _sabnzbdplus_
Any sensitive data has been scrubbed.

Line 808 is where the renamer kicks in. On line 809 the file name issue can be seen:

`2022-02-09 17:52:56,273::DEBUG::[sorting:304] Rename: /storage/TV/Quark.S01E01.Quark.1080p.WEB-DL.AAC2.0.H.264-squalor/Quark.S01E01.Quark.1080p.WEB-DL.AAC2.0.H.264-squalor.mkv to /storage/TV/Quark.S01E01.Quark.1080p.WEB-DL.AAC2.0.H.264-squalor/Quark\Quark.S01E01.Quark.1080p.WEB-DL.AAC2.0.H.264-squalor.mkv`

It seems like it's trying to add a Windows path seperator (\\) in between the show name and the file name. On Linux systems a backslash character is a valid character to use in a file name. So, the filename for the file from the above line is "Quark\Quark.S01E01.Quark.1080p.WEB-DL.AAC2.0.H.264-squalor.mkv" on my server. Samba seems to have an issue with passing file names that have a slash in them to remote systems (probably because Windows can't handle slashes in a file name), so it ends up passing along the file's short name instead.

---

This is the file on the server in a PuTTY SSH session:
![Screenshot from 2022-02-09 17-35-36](https://user-images.githubusercontent.com/8271857/153341704-96f85936-f73d-4c2b-807a-f895ac09e435.png)
It's located in the base TV download directory. For my system this is /storage/TV. The full file path for this file is /storage/TV/Quark\Quark.S01E01.Quark.1080p.WEB-DL.AAC2.0.H.264-squalor.mkv. In v3.3.1 the sorter puts the file in the /storage/TV/Quark directory properly.

---

This is the file as viewed in Nemo (Linux Mint's default file explorer) in short file name format:
![Screenshot from 2022-02-09 17-36-32](https://user-images.githubusercontent.com/8271857/153341729-01fc8208-63ee-4e7f-9508-c61554473532.png)
It displays the same way on my Windows 10 installation as well.

**Comment by Safihre** [comment]
Have you tried replacing the `\` with `/` in your Sorting setting?
As it's a valid Linux path, it's allowed indeed. Maybe that did change with the Sorting refactoring, I would have to validate that.

**Comment by Annihilator6000** [comment]
> Have you tried replacing the `\` with `/` in your Sorting setting? As it's a valid Linux path, it's allowed indeed. Maybe that did change with the Sorting refactoring, I would have to validate that.

I didn't try that until now - I hadn't even though about that because I've used that same sort string for quite a few years. It was a holdover from when I moved from running SAB on a Windows machine to running it on an Ubuntu server. 

I made that change in the sort string a few minutes ago and tested a download again. It worked properly with that change. Thank you!

I guess a possible QOL improvement/feature for what I encountered could be to replace any user-entered directory separators with Python's os.sep, or split and rejoin with os.path.join, or something similar for those of us that don't realize that we've put the incorrect separator in the string. That, of course, is assuming that something changed in the code that modified the behavior after v3.3.1.
